src/undo_log/undo_log.c

```diff
@@ -286,6 +286,8 @@ void UndoLog_Rollback
 
 	if(count == 0) return;
 
+	QueryCtx_LockForCommit();
+
```

Comment:
consider computing the undo ranges (lines 296-301) outside of the locked block

Reply:
I don't feel like this worth it it is just decrementing a counter in a loop